internal/kafka/internal/services/kafka.go

```diff
@@ -1337,20 +1383,63 @@ func (k *kafkaService) AssignBootstrapServerHost(kafkaRequest *dbapi.KafkaReques
 		return errors.NewWithCause(errors.ErrorGeneral, replaceErr, "generated host is not valid")
 	}
 
+	// If we enable KafkaTLS, the bootstrapServerHost should use the external domain name rather than the cluster domain
+	if k.kafkaConfig.EnableKafkaCNAMERegistration {
+		kafkaRequest.BootstrapServerHost = fmt.Sprintf("%s.%s", truncatedKafkaIdentifier, kafkaRequest.KafkasRoutesBaseDomainName)
+		return nil
+	}
+
 	clusterDNS, err := k.clusterService.GetClusterDNS(kafkaRequest.ClusterID)
 	if err != nil {
 		return errors.NewWithCause(errors.ErrorGeneral, err, "error retrieving cluster DNS")
 	}
 
 	clusterDNS = strings.Replace(clusterDNS, constants.DefaultIngressDnsNamePrefix, constants.ManagedKafkaIngressDnsNamePrefix, 1)
 
-	if k.kafkaConfig.EnableKafkaCNAMERegistration {
-		// If we enable KafkaTLS, the bootstrapServerHost should use the external domain name rather than the cluster domain
-		kafkaRequest.BootstrapServerHost = fmt.Sprintf("%s.%s", truncatedKafkaIdentifier, k.kafkaConfig.KafkaDomainName)
-	} else {
-		kafkaRequest.BootstrapServerHost = fmt.Sprintf("%s.%s", truncatedKafkaIdentifier, clusterDNS)
+	kafkaRequest.BootstrapServerHost = fmt.Sprintf("%s.%s", truncatedKafkaIdentifier, clusterDNS)
+	return nil
+}
+
+func (k *kafkaService) ManagedKafkasRoutesTLSCertificate(kafkaRequest *dbapi.KafkaRequest) error {
+	// Sets the KafkaRequest.KafkasRoutesBaseDomainName if the field is not set.
+	// When the certificate management strategy is manual, the kafka domain name given in the KafkaConfig is used.
+	// When the strategy is automatic, there are three cases:
+	// 1. We use the Kafka domain given in the Kafka config as the base domain for already prepared Kafkas
+	// 2. The base domain name is constructed by concatenating the kafka id and the kafka domain name given in the Kafka config for all standard Kafkas in preparing state i.e Kafkas that do not have the bootstrap server host already set
+	// 3. The base domain name is constructed by concatenating the "trial" and the kafka domain name given in the Kafka config for all developer Kafkas in preparing state i.e Kafkas that do not have the bootstrap server host already set
+	if !kafkaRequest.HasCertificateInfo() {
+		if !k.kafkaTLSCertificateManagementService.IsAutomaticCertificateManagementEnabled() {
+			kafkaRequest.KafkasRoutesBaseDomainName = k.kafkaConfig.KafkaDomainName
+		} else if kafkaRequest.Status == constants.KafkaRequestStatusAccepted.String() || kafkaRequest.Status == constants.KafkaRequestStatusPreparing.String() {
+			if kafkaRequest.IsADeveloperInstance() {
+				kafkaRequest.KafkasRoutesBaseDomainName = fmt.Sprintf("%s.%s", constants.TrialKafkasDomainShard, k.kafkaConfig.KafkaDomainName)
```

Comment:
why are developer instances handled differently?

Reply:
To keep our letsencrypt usage limits sane by reducing the consumption of whatever quota we've due to dev Kafkas being spinned up - during workshop etc
The reasoning is also capture in the ADR-90:   https://github.com/bf2fc6cc711aee1a0c2a/architecture/pull/84/files#diff-9ab93b157069c0c5414cc56ec41ae2ba8e6307f5ef2dd034ac269c8258b548f9

Reply:
Right, i knew there was a reason, but couldn't remember what it was. Thanks that makes sense. No individual revokation for developer instances then.

Reply:
> No individual revokation for developer instances then.

Exactly. It is considered as a shared cert.